src/librekey/key_store_g10.c

```diff
@@ -377,12 +377,15 @@ parse_pubkey(pgp_keydata_key_t *keydata, s_exp_t *s_exp, pgp_pubkey_alg_t alg)
 static bool
 parse_seckey(pgp_keydata_key_t *keydata, s_exp_t *s_exp, pgp_pubkey_alg_t alg)
 {
-    keydata->seckey.pubkey.version = PGP_V4;
-    keydata->seckey.pubkey.birthtime = time(NULL);
-    keydata->seckey.pubkey.duration = 0;
-    keydata->seckey.pubkey.alg = alg;
+    if (keydata->seckey.pubkey.version != PGP_V2 && keydata->seckey.pubkey.version != PGP_V3 &&
+        keydata->seckey.pubkey.version != PGP_V4) {
+        fprintf(stderr, "You should run parse_seckey only after parse_pubkey\n");
+        return false;
+    }
 
     keydata->seckey.s2k_usage = PGP_S2KU_NONE;
+    keydata->seckey.alg = PGP_SA_DEFAULT_CIPHER;
+    keydata->seckey.hash_alg = PGP_DEFAULT_HASH_ALGORITHM;
```

Comment:
These should probably just be 0, they're only really valid for .gpg encrypted keys (S2K).